rs/nervous_system/agent/src/pocketic_impl.rs

```diff
@@ -205,4 +211,37 @@ impl CallCanisters for PocketIc {
     fn caller(&self) -> Result<Principal, Self::Error> {
         Ok(Principal::anonymous())
     }
+
+    fn is_canister_stopped_error(&self, err: &Self::Error) -> bool {
+        match err {
+            PocketIcCallError::PocketIc(err) => {
+                [ErrorCode::CanisterStopped, ErrorCode::CanisterStopping].contains(&err.error_code)
+            }
+            _ => false,
+        }
+    }
+}
+
+impl ProgressNetwork for PocketIcAgent<'_> {
+    async fn progress(&self, duration: Duration) {
+        self.pocket_ic.progress(duration).await
+    }
+}
+
+impl ProgressNetwork for PocketIc {
+    async fn progress(&self, duration: Duration) {
+        // If PocketIC instance doesn't have a URL, thus it's not in the live mode and
+        // we can progress it using 'advance_time' method.
+        if self.url().is_none() {
+            self.tick().await;
+            self.advance_time(duration).await;
```

Comment:
I'd flip ticking and advancing time so that the advanced time is certified by the subsequent tick.